src/kafka_fdw.c

```diff
@@ -170,33 +201,64 @@ kafkaGetForeignPaths(PlannerInfo *root, RelOptInfo *baserel, Oid foreigntableid)
     KafkaEstimateCosts(root, baserel, fdw_private, &startup_cost, &total_cost, &run_cost);
 
     relation_close(relation, AccessShareLock);
+
     /*
      * Create a ForeignPath node and add it as only possible path.  We use the
      * fdw_private list of the path to carry the convert_selectively option;
      * it will be propagated into the fdw_private list of the Plan node.
      */
-
-    foreign_path = kafka_create_foreignscan_path(root,
-                                                 baserel,
-                                                 NULL, /* default pathtarget */
-                                                 baserel->rows,
-                                                 startup_cost,
-                                                 total_cost,
-                                                 NIL,  /* no pathkeys */
-                                                 NULL, /* no outer rel either */
-                                                 NULL, /* no extra plan */
-                                                 NIL);
+    foreign_path = (Path *)
```

Comment:
The comment above looks outdated.

Reply:
Removed comment and redundant `relation_open`/`relation_close`